analyses/snv-callers/compare_snv_callers_plots.Rmd

```diff
@@ -72,69 +63,148 @@ if (!("GGally" %in% installed.packages())) {
 `%>%` <- dplyr::`%>%`
 ```
 
-Set up output directories. 
+Set up directories. 
 
 ```{r}
-base_results_dir <- "results"
-base_plots_dir <- "plots"
+scratch_dir <- file.path("..", "..", "scratch")
+results_dir <- file.path("results", "comparison")
+plots_dir <- file.path("plots", "comparison")
+
+if (!dir.exists(results_dir)) {
+  dir.create(results_dir)
+}
+if (!dir.exists(plots_dir)) {
+  dir.create(plots_dir)
+}
 ```
 
-Make new directories for the comparison analysis.
+## Connect to database
+
+Connect to SQLite database.
 
 ```{r}
-consensus_results_dir <- file.path(base_results_dir, "consensus")
-consensus_plots_dir <- file.path(base_plots_dir, "consensus")
+# Start up connection
+con <- DBI::dbConnect(RSQLite::SQLite(), 
+                      file.path(scratch_dir, "testing_snv_db.sqlite"))
+```
 
-# Make caller specific plots folder
-if (!dir.exists(consensus_results_dir)) {
-  dir.create(consensus_results_dir)
-}
-# Make caller specific plots folder
-if (!dir.exists(consensus_plots_dir)) {
-  dir.create(consensus_plots_dir)
-}
+Note what columns we will join by.
+
+```{r}
+join_cols = c("Chromosome",
+              "Start_Position",
+              "Reference_Allele",
+              "Allele",
+              "Tumor_Sample_Barcode")
+```
+
+Set up tables from the database but only call the columns we need.
+
+```{r}
+strelka <- dplyr::tbl(con, "strelka") %>% 
+  dplyr::select(join_cols, "VAF")
+
+lancet <- dplyr::tbl(con, "lancet") %>% 
+  dplyr::select(join_cols, "VAF")
+
+mutect <- dplyr::tbl(con, "mutect") %>% 
+  dplyr::select(join_cols, "VAF") 
+
+vardict <- dplyr::tbl(con, "vardict") %>% 
+  dplyr::select(join_cols, "VAF") 
 ```
 
-## Import data 
+```{r}
+# Designate consensus table
+consensus <- dplyr::tbl(con, "consensus")
+```
 
-Load in the merged data.frames from `03-merge_callers.R` output. 
+Because `DBI` does not support full_join, we had to use a series of `left_join` and
+`union_all` calls in order to get a full join of all the callers. 
 
 ```{r}
-# Master VAF data.frame
-vaf_df <- readr::read_rds(file.path(consensus_results_dir, 
-                                    "all_callers_vaf.rds"))
+# This script will do the full join of the data for us
+source(file.path("util", "full_join_callers.R"))
+
+# Check that this looks right
+head(all_caller)
+```
 
-# Master TMB data.frame
-tmb_df <- readr::read_rds(file.path(consensus_results_dir, 
-                                    "all_callers_tmb.rds"))
+```{r}
+# Set up the data as a data.frame
+vaf_df <- all_caller %>% 
+  dplyr::select("VAF_lancet", "VAF_strelka", "VAF_vardict", "VAF_mutect") %>%
+  as.data.frame() %>%
+  dplyr::mutate(VAF_consensus = VAF_strelka)     
+ 
+# Which should be part of the consensus
+consensus_index <- which(apply(!is.na(vaf_df[, -3]), 1, sum) < 3)
```

Comment:
This may be a bit clearer:
```suggestion
consensus_index <- vaf_df %>%
  dplyr::select(-c(VAF_vardict, VAF_consensus)) %>%
  rowSums(is.na(.)) > 0
```

Reply:
I kept writing this, being like there is definitely a better way to do this but I don't know why I couldn't figure it out. This is good. Thanks.